File: Modules/Front/Credit.Front/Client/Components/CreditApplicationDto.cs
namespace Credit.Front.Client.Components;

public interface IWithCreditParams
{
    decimal Amount { get; init; }
    decimal CreditPeriodInMonths { get; init; }
}

public interface IWithDeclerations
{
    DeclarationDto Declaration { get; init; }
}

public interface IWithCustomerPersonal
{
    CustomerPersonalDto CustomerPersonalData { get; init; }
}

public interface IWithCreditApplicationState
{
    ApplicationState State { get; init; }
}

public record CustomerPersonalDto(string FirstName, string LastName, string Pesel);

public record DeclarationDto(decimal AverageNetMonthlyIncome);


public record ApplicationState(Level Level, DateTimeOffset Date, DateTimeOffset? ContractSigningDate, Decision Decision);

public enum Level
{
    ApplicationRegistered, DecisionGenerated, ContractSigned, ApplicationClosed
}

public enum Decision
{
    NotExists, Positive, Negative
}

File: Modules/Front/credit.front.next/src/app/(dashboard)/applications/(dto)/creditApplicationDto.ts
export interface WithCreditParams {
    amount: number;
    creditPeriodInMonths: number;
}

export interface WithDeclerations {
    declaration: DeclarationDto;
}

export interface WithCustomerPersonal {
    customerPersonalData: CustomerPersonalDto;
}

export interface CustomerPersonalDto {
    firstName: string;
    lastName: string;
    documentId: string;
}

export interface DeclarationDto {
    averageNetMonthlyIncome: number;
}

export interface WithCreditApplicationState{
    state: ApplicationState;
}

export interface ApplicationState{
    level: "ApplicationRegistered" | "DecisionGenerated" | "ContractSigned" | "ApplicationClosed";
    date: Date,
    contractSigningDate: Date,
    decision: "NotExists" | "Positive" | "Negative"
}